src/python_testing/TC_FAN_2_1.py

```diff
@@ -0,0 +1,171 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+# See https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md#defining-the-ci-test-arguments
+# for details about the block below.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${AIR_PURIFIER_APP}
+#     app-args: --discriminator 1234 --KVS kvs1 --trace_file json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import logging
+from typing import Any
+
+import chip.clusters as Clusters
+from chip.clusters import ClusterObjects as ClusterObjects
+from chip.interaction_model import Status
+from chip.testing.matter_asserts import is_valid_uint_value
+from chip.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+logger = logging.getLogger(__name__)
+
+
+class Uint8Type(int):
+    pass
+
+
+class TC_FAN_2_1(MatterBaseTest):
+    def desc_TC_FAN_2_1(self) -> str:
+        return "[TC-FAN-2.1] Mandatory functionality with DUT as Server"
+
+    def steps_TC_FAN_2_1(self):
+        return [TestStep(1, "[FC] Commissioning already done.", is_commissioning=True),
+                TestStep(2, "[FC] TH checks for support of the Auto feature.",
+                         "Save result for future use."),
+                TestStep(3, "[FC] TH reads from the DUT the FanModeSequence attribute.",
+                         "Verify that the DUT response contains a FanModeSequenceEnum with value between 0 and 5 inclusive. If Auto is not supported, verify that the FanModeSequence attribute is a valid sequence (non-Auto). If Auto is supported, verify that the FanModeSequence attribute is a valid sequence (Auto or non-Auto)."),
```

Comment:
if AUTO is supported, the non-auto sequences are disallowed - see appclusters 4.4.5.6, non-auto sequences are [!AUT].a